drv/hisi_sec.c

```diff
@@ -756,3 +759,303 @@ static struct wd_digest_driver hisi_digest_driver = {
 };
 
 WD_DIGEST_SET_DRIVER(hisi_digest_driver);
+
+static int fill_aead_bd2_alg(struct wd_aead_msg *msg,
+	struct hisi_sec_sqe *sqe)
+{
+	__u8 c_key_len = 0;
+	__u32 d_alg = 0;
+	int ret = 0;
+
+	switch (msg->calg) {
+	case WD_CIPHER_SM4:
+		sqe->type2.c_alg = C_ALG_SM4;
+		sqe->type2.icvw_kmode = CKEY_LEN_SM4 << SEC_CKEY_OFFSET;
+		break;
+	case WD_CIPHER_AES:
+		sqe->type2.c_alg = C_ALG_AES;
+		if (msg->ckey_bytes == AES_KEYSIZE_128) {
+			c_key_len = CKEY_LEN_128BIT;
+		} else if (msg->ckey_bytes == AES_KEYSIZE_192) {
+			c_key_len = CKEY_LEN_192BIT;
+		} else if (msg->ckey_bytes == AES_KEYSIZE_256) {
+			c_key_len = CKEY_LEN_256BIT;
+		} else {
+			WD_ERR("failed to check AES key size!\n");
+			return -EINVAL;
+		}
+		sqe->type2.icvw_kmode = (__u16)c_key_len << SEC_CKEY_OFFSET;
```

Comment:
The upper code if in a function named xxx_get_ken_len(xxx)  is better.

Reply:
Yes，in the cipher algorithm, there is an function named get_aes_c_key_len();